### PR TITLE
initial board page commit

### PR DESCRIPTION
**Pull request by KevinLu9**
only contains the draggable cards and hardcoded mock data.

### CODE REVIEW

**Comment by KevinLu9** [comment]
[Create Basic Structure of board page](https://trello.com/c/vhqgkvBR/34-create-basic-structure-of-board-page)

**Comment by KevinLu9** [comment]
boardpage

File: frontend/src/components/Board/Board.tsx [hunk 1]
```diff
@@ -1,7 +1,327 @@
...
+      setMyColumns(newColumn);
+    }
+  }
+
+  return (
+    <>
+      <div className="bg-gray-600 p-2 h-full overflow-x-auto">
+        <DragDropContext onDragEnd={onDragEnd}>
+          <div className="h-full flex flex-row">
+            {myColumns.map((col: IColumn, index: number) => (
```
Comment (Imeth97) [inline frontend/src/components/Board/Board.tsx:294]: when we map over a list of components, could we use that repeater.tsx function I made? should clean up the code a lot more & has built in deletion handling etc. ofc feel free to add to the functionality in it

File: frontend/src/components/Board/Board.tsx [hunk 2]
```diff
@@ -1,7 +1,327 @@
...
+          <Button className="justify-self-end" size="small">
+            Edit
+          </Button>
+        </CardActions>
+      </Card>
+    </>
+  );
+}
+
+function TaskColumn(props: IColumnProp) {
```
Comment (Imeth97) [inline frontend/src/components/Board/Board.tsx]: lets try and directly destructure props, cleaner code, avoids repetitively calling "props." everywhere

File: frontend/src/components/Board/Board.tsx [hunk 3]
```diff
@@ -1,7 +1,327 @@
...
+  created_at: new Date().toDateString(),
+  user_id: "1111",
+  team_id: "2222",
+  name: "FIT3162 Board",
+  description:
+    "This board is used to monitor the progress of the FIT3162 CS_20 Project.",
+};
+
+// list table
+const columns: IColumn[] = [
```
Comment (Imeth97) [inline frontend/src/components/Board/Board.tsx:33]: lets put the types in a seperate d.ts file. cleans up & we dont need to import types too if we use d.ts

**Review by Imeth97 — COMMENTED** [review]
great stuff overall, just a few minor things related to code quality only

File: frontend/src/components/Board/Board.tsx [hunk 4]
```diff
@@ -1,7 +1,339 @@
...
+//     user_creator: "1",
+//     user_assigned: "1",
+//     title: "Get to know the team!",
+//     description: "Do ice breaker activities to understand the team better.",
+//     deadline: new Date().toDateString(),
+//     created_at: new Date().toDateString(),
+//   },
+// ];
+
+const team = {
```
Comment (mursal-ashraf) [inline frontend/src/components/Board/Board.tsx:145]: Are `team` and `members` gonna be used?

File: frontend/src/components/Board/Board.tsx [hunk 5]
```diff
@@ -1,7 +1,339 @@
...
+      </div>
+    </>
+  );
+}
+
+function TaskBoard() {
+  const [myColumns, setMyColumns] = useState(columns);
+
+  // Handles moving cards
+  function onDragEnd(result: any) {
```
Comment (mursal-ashraf) [inline frontend/src/components/Board/Board.tsx]: Getting a lint warning for the `any` here, any types that you can use from library here?

File: frontend/src/components/Board/Board.tsx [hunk 6]
```diff
@@ -1,7 +1,339 @@
-import React from 'react';
+import { useState } from 'react';
+// import { AccountCircle } from "@mui/icons-material";
```
Comment (mursal-ashraf) [inline frontend/src/components/Board/Board.tsx:2]: Is this import gonna be used?

File: frontend/src/components/Board/Board.tsx [hunk 7]
```diff
@@ -1,7 +1,339 @@
...
+    const columnDestination: IColumn | undefined = columns?.find(
+      (c) => c.id == result?.destination?.droppableId
+    );
+    const cardToMove: ICard | undefined = columnSource?.cards?.find(
+      (c) => c.id == result?.draggableId
+    );
+    if (cardToMove && columnSource && columnDestination) {
+      cardToMove.list_id = result?.destination?.droppableId;
+      // If moving cards in the same column
+      if (result?.source?.droppableId == result?.destination?.droppableId) {
```
Comment (mursal-ashraf) [inline frontend/src/components/Board/Board.tsx:260]: I'm guessing the order of splicing columnSource or columnDestination matters? (It's the only difference I can tell between the if and else)

Comment (KevinLu9) [reply frontend/src/components/Board/Board.tsx:260]: Thats my 5am brain thinking, but its what we need for the difference between switching cards between columns and in the same column.

File: frontend/src/components/Board/Board.tsx (hunk 6 above)
Comment (KevinLu9) [reply frontend/src/components/Board/Board.tsx:2]: was going to for the member icons, but left it out for now.

File: frontend/src/components/Board/Board.tsx (hunk 4 above)
Comment (KevinLu9) [reply frontend/src/components/Board/Board.tsx:145]: yea, to show who else has access to the board and assigning cards to people

**Review by mursal-ashraf — APPROVED** [review]

File: frontend/src/components/Board/Board.tsx (hunk 2 above)
Comment (KevinLu9) [reply frontend/src/components/Board/Board.tsx]: yea, my bad, i initially had alot more stuff in that props, but I removed them later on.

Comment (KevinLu9) [reply frontend/src/components/Board/Board.tsx]: opps nvm, i read ur comment wrong, ill do that

**Review by KevinLu9 — COMMENTED** [review]
just for simplicity, could we stick to the default prettier? I think it uses double quotes by default.

**Pull request merged**